pkg/daemon/update.go

```diff
@@ -618,26 +679,18 @@ func (dn *Daemon) updateSSHKeys(newUsers []ignv2_2types.PasswdUser) error {
 		// Double checking that we are only writing SSH Keys for user "core"
 		return fmt.Errorf("Expecting user core. Got %s instead", newUsers[0].Name)
 	}
-	sshDirPath := filepath.Join("/home", newUsers[0].Name, ".ssh")
-	// we are only dealing with the "core" User at this time, so only dealing with the first entry in Users[]
-	glog.Infof("Writing SSHKeys at %q", sshDirPath)
-	if err := dn.fileSystemClient.MkdirAll(filepath.Dir(sshDirPath), os.FileMode(0600)); err != nil {
-		return fmt.Errorf("Failed to create directory %q: %v", filepath.Dir(sshDirPath), err)
-	}
-	glog.V(2).Infof("Created directory: %s", sshDirPath)
 
-	authkeypath := filepath.Join(sshDirPath, "authorized_keys")
+	// we're also appending all keys for any user to core, so for now
```

Comment:
Not quite: "we are appending all new SSHAuthorizedKeys of user core and passing to atomicallyWriteSSHKeys to write."

Reply:
mmm https://github.com/openshift/machine-config-operator/pull/401/files#diff-06961b075f1753956d802ba954d2cfb5R686 seems to suggest that we're ranging through every user in Users and grabbind their ssh keys as well

Reply:
no we are ranging over the SSHAuthorizedKeys

Reply:
well, no the last one actually

Reply:
there's no check around `len(newUsers)` so that range can be wrong right?

Reply:
We are looking at the last user in new Users(as per Ignition) and ranging over its SSHKeys. There is a len check in updateSSHKeys the first or second line.

Reply:
> no we are ranging over the SSHAuthorizedKeys

 yeah, I was wrong it's `SSHAuthorizedKeys`. My point being though, we're checking for the very first user to be `core` but if `newUsers` contains more than 1 entry, we're ending up appending keys from `len(newUsers)-1` User, not `core` (is it possible?)

Reply:
alrighty, thanks, just saw your previous comment :smile:

Reply:
Yeah I have a cleanup PR taking that erroneous [0] out :)

Reply:
no wait, the len is for newUsers to be empty, but what if it has more than 1 user? we'd fall into https://github.com/openshift/machine-config-operator/pull/401#discussion_r256176416

Reply:
I think it's prob better to go over this with you outside of this pr discussion thread :)

tho i'd like this pr to wait until my cleanup pr hits and you can rebase on it.  though it sounds like this one is going to be held anyway?

Reply:
well, sure, I can def wait for you PR to land first. As for holding this up, for now yea, but it's still something we need to have imo (nodes can go down, lose connection and stuff like that)